pkg/commands/image/patch_test.go

```diff
@@ -306,7 +306,9 @@ func testImagePatchCommand(t *testing.T, when spec.G, it spec.S) {
 		})
 
 		it("can output resources in yaml and does not wait", func() {
```

Comment:
@djoyahoy raised this issue a while back and at the time the plan was to go with whatever kubectl did. But, when I tried it with kubectl, what I realized was that `kp patch` and `kubectl patch` are very different in behaviour
- `kubectl patch` command takes the patch file as input and so for --output with patch it outputs the complete resource yaml
  ```
  $ k patch rs kubernetes-bootcamp-69fbc6f4cf -p '{"spec": {"replicas": 3}}' --dry-run --output yaml
  W1005 14:35:09.672041   23199 helpers.go:535] --dry-run is deprecated and can be replaced with --dry-run=client.
  apiVersion: apps/v1
  kind: ReplicaSet
  metadata:
    name: kubernetes-bootcamp-69fbc6f4cf
    ....
  spec:
    replicas: 3
    ....
  ```

- whereas with `kp patch` the user provides new values using flags and the kp cli generates and applies the patch. So, in this scenario, we could output the patch file so that the user can apply a patch using kubectl. But providing the updated full resource file is also functional, since it can be applied using kubectl.

The other thing to consider would be that if we go down the path of providing patch output instead of full resource output, the create command behaviour will be different from patch as expected, but the save command would also become unpredictable - either outputting full resource or the patch file. Whereas as of now, all 3 commands behave the same.

Reply:
I think, this stuff is new (and better). 
```
W1005 14:35:09.672041   23199 helpers.go:535] --dry-run is deprecated and can be replaced with --dry-run=client.
```
Increasing validation scope of --dry-run to server side too instead of limiting to client-side validation only, like we do. lot more work involved to do this at server side too.

Reply:
> save command would also become unpredictable

very true... I think we should change `kp save` to create or update, never patch. for the reasons you mentioned about output and the `kp save` command probably won't be used as much as a patch command and it's primary use case to me is likely as an idempotent (CI) command.

Reply:
> kubectl patch command takes the patch file as input and so for --output with patch it outputs the complete resource yaml

I must have missed this, I think this is a good enough reason to do the same.